Blocks/SS/pubsub.h

```diff
@@ -121,21 +124,61 @@ class StreamSubscriber : public GenericStreamSubscriber<ENTRY>, public EntrySubs
   using EntrySubscriber<IMPL, ENTRY>::EntrySubscriber;
 };
 
+// For `static_assert`-s. Must `decay<>` for template xvalue references support.
 template <typename T>
 struct IsSubscriber {
-  static constexpr bool value = std::is_base_of<GenericSubscriber, T>::value;
+  static constexpr bool value = std::is_base_of<GenericSubscriber, current::decay<T>>::value;
 };
 
 template <typename T, typename E>
 struct IsEntrySubscriber {
-  static constexpr bool value = std::is_base_of<GenericEntrySubscriber<E>, T>::value;
+  static constexpr bool value =
+      std::is_base_of<GenericEntrySubscriber<current::decay<E>>, current::decay<T>>::value;
 };
 
 template <typename T, typename E>
 struct IsStreamSubscriber {
-  static constexpr bool value = std::is_base_of<GenericStreamSubscriber<E>, T>::value;
+  static constexpr bool value =
+      std::is_base_of<GenericStreamSubscriber<current::decay<E>>, current::decay<T>>::value;
 };
 
+namespace impl {
+
+template <typename TYPE_SUBSCRIBED_TO, typename STREAM_UNDERLYING_VARIANT>
+struct PassEntryToSubscriberOrSkipItImpl {
+  static_assert(TypeListContains<typename STREAM_UNDERLYING_VARIANT::T_TYPELIST, TYPE_SUBSCRIBED_TO>::value,
+                "Subscribing to the type different from the underlying type requires the underlying type"
+                " to be a `Variant<>` containing the subscribed to type as an option.");
+
+  template <typename F, typename E>
+  static EntryResponse Dispatch(F&& f, E&& entry, idxts_t current, idxts_t last) {
+    static_assert(IsEntrySubscriber<F, TYPE_SUBSCRIBED_TO>::value, "");
+    const E& entry_cref = entry;
+    if (Exists<TYPE_SUBSCRIBED_TO>(entry_cref)) {
```

Comment:
It would accept it, yes, but semantically I'm worried about using `entry` an rvalue reference twice. In this case it would work, but to an unsavvy observer it might look like `x; f(std::move(x)); g(std::move(x));`, or even `x; y = std::move(x); z = std::move(x);`, which does not seem right.